pkg/util/errors.go

```diff
@@ -16,14 +16,14 @@ limitations under the License.
 
 package util
 
-// ErrKeyNotFound is returned when requested key in omap is not found
+// ErrKeyNotFound is returned when requested key is not found
```

Comment:
I am not comfortable in overloading this error for missing pools. I would prefer a separate error code, although dealing with an additional error code in the callers is more code, it keeps the error codes cleaner.